PDF-Import not working and crash with error

<img width="897" height="240" alt="Image" src="https://github.com/user-attachments/assets/122c5ebb-22fc-4ef8-ab50-3ffa67dd43d8" />


```
Fri Oct 17 17:33:04 CEST 2025
InvocationTargetException

java.lang.reflect.InvocationTargetException
	at org.eclipse.jface.operation.ModalContext.run(ModalContext.java:396)
	at org.eclipse.jface.dialogs.ProgressMonitorDialog.run(ProgressMonitorDialog.java:470)
	at name.abuchen.portfolio.ui.handlers.ImportPDFHandler.runImportWithFiles(ImportPDFHandler.java:178)
	at name.abuchen.portfolio.ui.dnd.ImportFromFileDropAdapter.lambda$0(ImportFromFileDropAdapter.java:73)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:132)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4177)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3793)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1151)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1042)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:153)
	at org.eclipse.e4.ui.internal.workbench.swt.E4Application.start(E4Application.java:165)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:208)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:143)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:109)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:439)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:271)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:668)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:605)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1481)
Caused by: java.lang.ArrayIndexOutOfBoundsException: arraycopy: last source index 32 out of bounds for byte[16]
	at java.base/java.lang.System.arraycopy(Native Method)
	at org.apache.pdfbox.pdmodel.encryption.StandardSecurityHandler.computeEncryptedKey(StandardSecurityHandler.java:571)
	at org.apache.pdfbox.pdmodel.encryption.StandardSecurityHandler.computeUserPassword(StandardSecurityHandler.java:609)
	at org.apache.pdfbox.pdmodel.encryption.StandardSecurityHandler.isUserPassword(StandardSecurityHandler.java:805)
	at org.apache.pdfbox.pdmodel.encryption.StandardSecurityHandler.prepareForDecryption(StandardSecurityHandler.java:216)
	at org.apache.pdfbox.pdmodel.encryption.StandardSecurityHandler.decryptDocument(StandardSecurityHandler.java:156)
	at org.apache.pdfbox.pdmodel.PDDocument.openProtection(PDDocument.java:1646)
	at org.apache.pdfbox.pdmodel.PDDocument.decrypt(PDDocument.java:974)
	at name.abuchen.portfolio.pdfbox1.PDFBox1Adapter.convertToText(PDFBox1Adapter.java:20)
	at name.abuchen.portfolio.datatransfer.pdf.PDFInputFile.convertLegacyPDFtoText(PDFInputFile.java:78)
	at name.abuchen.portfolio.datatransfer.pdf.PDFImportAssistant.run(PDFImportAssistant.java:184)
	at name.abuchen.portfolio.ui.handlers.ImportPDFHandler.lambda$3(ImportPDFHandler.java:175)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:124)
```


[4434.13_PLV_finvesto_Depot.pdf](https://github.com/user-attachments/files/22974551/4434.13_PLV_finvesto_Depot.pdf)

Testfile... yes i know... i know... its a testfile

--- a/name.abuchen.portfolio.pdfbox1/src/name/abuchen/portfolio/pdfbox1/PDFBox1Adapter.java
+++ b/name.abuchen.portfolio.pdfbox1/src/name/abuchen/portfolio/pdfbox1/PDFBox1Adapter.java
@@ -25,7 +25,7 @@ public class PDFBox1Adapter
             textStripper.setSortByPosition(true);
             return textStripper.getText(document);
         }
-        catch (CryptographyException e)
+        catch (RuntimeException | CryptographyException e)
         {
             throw new IOException(e);
         }

--- a/name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/handlers/CreateTextFromPDFHandler.java
+++ b/name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/handlers/CreateTextFromPDFHandler.java
@@ -2,9 +2,7 @@ package name.abuchen.portfolio.ui.handlers;
 
 import java.io.File;
 import java.io.IOException;
-import java.util.Arrays;
 import java.util.Random;
-import java.util.stream.Collectors;
 
 import jakarta.inject.Named;
 
@@ -19,9 +17,6 @@ import org.eclipse.swt.widgets.Composite;
 import org.eclipse.swt.widgets.FileDialog;
 import org.eclipse.swt.widgets.Shell;
 import org.eclipse.swt.widgets.Text;
-
-import com.github.difflib.DiffUtils;
-import com.github.difflib.UnifiedDiffUtils;
 
 import name.abuchen.portfolio.datatransfer.pdf.PDFInputFile;
 import name.abuchen.portfolio.money.CurrencyUnit;
@@ -56,26 +51,9 @@ public class CreateTextFromPDFHandler
             var extractedText = inputFile.getText();
             var pdfBoxVersion = inputFile.getPDFBoxVersion();
 
-            // check if the extracted text changed with the PDFBox version
-            inputFile.convertLegacyPDFtoText();
-            var legacyText = inputFile.getText();
-            var legacyPdfBoxVersion = inputFile.getPDFBoxVersion();
-            var isDifferent = !extractedText.equals(legacyText);
-
-            if (isDifferent)
-            {
-                var patch = DiffUtils.diff(legacyText, extractedText, null);
-                var unifiedDiff = UnifiedDiffUtils.generateUnifiedDiff(legacyPdfBoxVersion, pdfBoxVersion,
-                                Arrays.asList(legacyText.split("\n")), patch, 0);
-                PortfolioPlugin.info(
-                                inputFile.getName() + "\n\n" + unifiedDiff.stream().collect(Collectors.joining("\n")));
-            }
-
             StringBuilder textBuilder = new StringBuilder();
             textBuilder.append("```").append("\n");
-            textBuilder.append("PDFBox Version: ").append(pdfBoxVersion)
-                            .append(isDifferent ? " != " + legacyPdfBoxVersion : "") //
-                            .append("\n");
+            textBuilder.append("PDFBox Version: ").append(pdfBoxVersion).append("\n");
             textBuilder.append("Portfolio Performance Version: ")
                             .append(PortfolioPlugin.getDefault().getBundle().getVersion().toString()) //
                             .append("\n");

--- a/name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/PDFImportAssistant.java
+++ b/name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/PDFImportAssistant.java
@@ -181,22 +181,30 @@ public class PDFImportAssistant
 
                 if (!extracted)
                 {
-                    inputFile.convertLegacyPDFtoText();
-                    for (Extractor extractor : extractors)
+                    try
                     {
-                        var items = extractor.extract(securityCache, inputFile, warnings);
-
-                        if (!items.isEmpty())
+                        inputFile.convertLegacyPDFtoText();
+                        for (Extractor extractor : extractors)
                         {
-                            extracted = true;
-                            itemsByExtractor.computeIfAbsent(extractor, e -> new ArrayList<Item>()).addAll(items);
-                            break;
+                            var items = extractor.extract(securityCache, inputFile, warnings);
+
+                            if (!items.isEmpty())
+                            {
+                                extracted = true;
+                                itemsByExtractor.computeIfAbsent(extractor, e -> new ArrayList<Item>()).addAll(items);
+                                break;
+                            }
+                        }
+
+                        if (extracted)
+                        {
+                            PortfolioLog.info("PDF successfully imported with PDFBox 1.8.x " + inputFile.getName()); //$NON-NLS-1$
                         }
                     }
-
-                    if (extracted)
+                    catch (IOException ignore)
                     {
-                        PortfolioLog.info("PDF successfully imported with PDFBox 1.8.x " + inputFile.getName()); //$NON-NLS-1$
+                        // ignore if the file cannot be read by PDFBox Version 1
+                        PortfolioLog.error(ignore);
                     }
                 }
 

